src/js/index.js

```diff
@@ -503,26 +506,49 @@
                 bip32RootKey = bitcoinjs.bitcoin.HDNode.fromBase58(rootKeyBase58, n);
                 return;
             }
-            catch (e) {}
+            catch (e) {
+                try {
+                    bip32RootKey = groestlcoinjs.HDNode.fromBase58(rootKeyBase58, n);
+                    return;
+                }
+                catch(e) {}
+            }
             // try parsing using p2wpkh params
             if ("p2wpkh" in n) {
                 try {
                     bip32RootKey = bitcoinjs.bitcoin.HDNode.fromBase58(rootKeyBase58, n.p2wpkh);
                     return;
                 }
-                catch (e) {}
+                catch (e) {
+                    try {
+                        bip32RootKey = groestlcoinjs.HDNode.fromBase58(rootKeyBase58, n.p2wpkh);
+                        return;
+                    }
+                    catch(e) {}
+                }
             }
             // try parsing using p2wpkh-in-p2sh network params
             if ("p2wpkhInP2sh" in n) {
                 try {
                     bip32RootKey = bitcoinjs.bitcoin.HDNode.fromBase58(rootKeyBase58, n.p2wpkhInP2sh);
                     return;
                 }
-                catch (e) {}
+                catch (e) {
+                    try {
+                        bip32RootKey = groestlcoinjs.HDNode.fromBase58(rootKeyBase58, n.p2wpkhInP2sh);
+                        return;
+                    }
+                    catch(e) {}
+                }
             }
         }
         // try the network params as currently specified
-        bip32RootKey = bitcoinjs.bitcoin.HDNode.fromBase58(rootKeyBase58, network);
+        try {
+            bip32RootKey = bitcoinjs.bitcoin.HDNode.fromBase58(rootKeyBase58, network);
+        }
+        catch (e) {
+            bip32RootKey = groestlcoinjs.HDNode.fromBase58(rootKeyBase58, network);
+        }
     }
 
     function calcBip32ExtendedKey(path) {
```

Comment:
This method does get messy with all the exception handling. We could make a separate method that handles only Groestlcoin and then call it if isGRS()==true

Reply:
This change was made.